src/include/catalog/catalog.h

```diff
@@ -0,0 +1,314 @@
+#pragma once
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+#include "catalog/catalog_defs.h"
+#include "catalog/catalog_sql_table.h"
+#include "catalog/database_handle.h"
+#include "catalog/tablespace_handle.h"
+#include "common/strong_typedef.h"
+#include "loggers/catalog_logger.h"
+#include "storage/sql_table.h"
+
+namespace terrier::catalog {
+
+class DatabaseHandle;
+class TablespaceHandle;
+class SettingsHandle;
+
+/**
+ * Schema column for used/unused schema rows.
+ */
+struct SchemaCol {
```

Comment:
Correct. It is only used when creating catalog tables. Will comment.

Stating probable direction. We discussed in a team meeting adding default values into the schema stored with the sql table. The comment in schema.h implies we might replace the sql table schema with the schema from the catalog. This is possible, but extracting the schema information from the catalog is much more work. So, in my opinion, possible approach is to continue:

- schema stored in the catalog (as expected)
- retain the schema in the sql table layer, with default value extension. This can be viewed as a cached version of the schema, that is easier to interpret.
- need to ensure that such a scheme works correctly with the schema versioning project.

Reply:
Comment added, clarifying usage.

Reply:
When you say "the schema from the catalog", do you just mean the information in pg_attribute and pg_attrdef? I think even though we retrieve the schema information for the user tables from the catalog, in our own code base, we still need some class to store the schema information for the catalog tables themselves. It's kind of like a chicken-and-egg problem. Without some sort of schema representation, we cannot even define pg_attribute and pg_attrdef, right?